test/test-lex.cpp

```diff
@@ -113,6 +113,149 @@ TEST_F(test_lex, lex_block_comments) {
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"******"), u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"*/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 0, u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"foo(a, true * direct */);"_sv);
+    lexer l(&input, &v);  // identifier
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::left_paren);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::comma);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::kw_true);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::right_paren);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::semicolon);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+    l.skip();
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"foo(a, true * direct "),
+                                    u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"**/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 1, u8"*/"))));
+  }
+}
+
+TEST_F(test_lex, lex_unopened_block_comment2) {
+  {
+    error_collector v;
+    padded_string input(u8"3; */hello"_sv);
+    lexer l(&input, &v);  // number
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();  // semicolon
+    EXPECT_EQ(l.peek().type, token_type::semicolon);
+    l.skip();  // identifier
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"3; "), u8"*/"))));
+  }
+}
+
+TEST_F(test_lex, lex_unopened_block_comment_with_regexp_literal) {
+  {
+    error_collector v;
+    padded_string input(u8"3; */ hello/"_sv);
```

Comment:
The name of this test is misleading. It's not an unopened block comment.

How about this?
```suggestion
// The leading */ should not be interpreted as the end of a block comment.
TEST_F(test_lex, lex_regexp_literal_starting_with_star_slash) {
  {
    error_collector v;
    padded_string input(u8"3 */ hello/"_sv);
```